src/System.Net.Primitives/src/System/Net/IPv4AddressHelper.cs

```diff
@@ -15,7 +15,7 @@ internal static class IPv4AddressHelper
         private const int NumberOfLabels = 4;
 
         // Only called from the IPv6Helper, only parse the canonical format
-        internal static unsafe int ParseHostNumber(string str, int start, int end)
+        internal static unsafe int ParseHostNumber(char* str, int start, int end)
```

Comment:
Ditto on using `ReadOnlySpan<char>` unless there's a strong reason to avoid it.  Will stop commenting on this, but it applies throughout.